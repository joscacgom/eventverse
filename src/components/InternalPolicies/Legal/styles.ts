import styled from 'styled-components'

export const LegalContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
    width: 100%;
    height: 100%;
    padding: 0 20px;
    text-align: center;
    font-size: 1.5rem;
    margin-top: 20px;
    & > *:first-child {
        margin: 0 auto;
    }
`
export const LegalTitle = styled.h1`
    width: auto; 
    height: auto; 
    font-weight: 600;
    font-style: normal;
    font-family: "Inter-SemiBold", "Inter", sans-serif;
    color: #000000;
    font-size: 36px;

    @media (max-width: 768px) {
        font-size: 24px;
    }
`

export const LegalText = styled.p`
    margin-bottom: 15px;
    font-size: 1.2rem;
    text-align: justify;

    @media (max-width: 768px) {
        font-size: 0.8rem;
        text-align: justify;
    }
`

export const LegalParagraph = styled.p`
    margin-bottom: 15px;
    font-size: 1.2rem;
`

export const LegalHeader = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
     & > *:first-child {
        margin-bottom: 15px;
    }

`

export const LegalBody = styled.p`
    margin-bottom: 15px;
    font-size: 1.2rem;
`
