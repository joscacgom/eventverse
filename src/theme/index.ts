import { ThemeContext } from 'styled-components'

import { useContext } from 'react'

const nord = {
  black0: '#222222',
  black1: '#434c5e',

  gray0: '#FCFCFC',
  gray1: '#ECEFF4',
  gray2: '#575757',
  gray3: '#B3B3B3',

  red: '#D65151',
  blue: '#0099FF',
  green: '#539362',
  white: '#FFFFFF'
}

const theme = {
  nord
}

export type Theme = typeof theme;

export default theme as Theme

declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}

export const useTheme = () => useContext(ThemeContext)
