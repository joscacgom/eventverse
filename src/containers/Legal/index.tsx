import React from 'react'
import { Legal as LegalComponent } from '@/components/InternalPolicies'
import { MainContainer } from './styles'
const Legal = () => {
  return (
    <MainContainer>
     <LegalComponent />
    </MainContainer>
  )
}

export default Legal
