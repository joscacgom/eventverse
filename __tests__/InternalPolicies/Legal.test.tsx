import React from 'react'
/* eslint-disable no-undef */
import 'jest-styled-components'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import theme from '@/theme'
import { ThemeProvider } from 'styled-components'

import Legal from '@/components/InternalPolicies/Legal'

describe('Legal Component', () => {
  it('renders without error', () => {
    render(
        <ThemeProvider theme={theme}>
            <Legal />
       </ThemeProvider>)
  }
  )
  it('displays the legal title', () => {
    render(<Legal />)
    const legalTitle = screen.getByText('Aviso Legal')
    expect(legalTitle).toBeInTheDocument()
  })

  it('displays the privacy and data treatment section', () => {
    render(<ThemeProvider theme={theme}>
            <Legal />
        </ThemeProvider>)
    const privacySection = screen.getByText('Privacidad y tratamiento de datos')
    expect(privacySection).toBeInTheDocument()
  })

  it('renders the legal text with the correct content', () => {
    render(<ThemeProvider theme={theme}>
            <Legal />
        </ThemeProvider>)
    const legalText = screen.getByText(
      'EventVerse, responsable del sitio web, en adelante RESPONSABLE, pone a disposición de los usuarios el presente documento, con el que pretende dar cumplimiento a las obligaciones dispuestas en la Ley 34/2002, de 11 de julio, de Servicios de la Sociedad de la Información y del Comercio Electrónico (LSSICE), así como informar a todos los usuarios del sitio web respecto a cuáles son las condiciones de uso. Toda persona que acceda a este sitio web asume el papel de usuario, comprometiéndose a la observancia y cumplimiento riguroso de las disposiciones aquí dispuestas, así como a cualquier otra disposición legal que fuera de aplicación. EventVerse, se reserva el derecho de modificar cualquier tipo de información que pudiera aparecer en el sitio web, sin que exista obligación de preavisar o poner en conocimiento de los usuarios dichas obligaciones, entendiéndose como suficiente con la publicación en el sitio web de EventVerse'
    )
    expect(legalText).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { container } = render(<ThemeProvider theme={theme}>
            <Legal />
        </ThemeProvider>)
    expect(container).toMatchSnapshot()
  })
})
