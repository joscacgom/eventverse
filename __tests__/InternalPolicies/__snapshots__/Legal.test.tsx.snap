// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legal Component matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  padding: 0 20px;
  text-align: center;
  font-size: 1.5rem;
  margin-top: 20px;
}

.c0 > *:first-child {
  margin: 0 auto;
}

.c2 {
  width: auto;
  height: auto;
  font-weight: 600;
  font-style: normal;
  font-family: "Inter-SemiBold","Inter",sans-serif;
  color: #000000;
  font-size: 36px;
}

.c4 {
  margin-bottom: 15px;
  font-size: 1.2rem;
  text-align: justify;
}

.c5 {
  margin-bottom: 15px;
  font-size: 1.2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 > *:first-child {
  margin-bottom: 15px;
}

.c3 {
  margin-bottom: 15px;
  font-size: 1.2rem;
}

@media (max-width:768px) {
  .c2 {
    font-size: 24px;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 0.8rem;
    text-align: justify;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Aviso Legal
      </h1>
    </div>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Ley de los servicios de la sociedad de la información (LSSI) 
      </h1>
    </div>
    <p
      class="c3"
    >
      <p
        class="c4"
      >
        <p
          class="c5"
        >
          EventVerse, responsable del sitio web, en adelante RESPONSABLE, pone a disposición de los usuarios el presente documento, con el que pretende dar cumplimiento a las obligaciones dispuestas en la Ley 34/2002, de 11 de julio, de Servicios de la Sociedad de la Información y del Comercio Electrónico (LSSICE), así como informar a todos los usuarios del sitio web respecto a cuáles son las condiciones de uso. Toda persona que acceda a este sitio web asume el papel de usuario, comprometiéndose a la observancia y cumplimiento riguroso de las disposiciones aquí dispuestas, así como a cualquier otra disposición legal que fuera de aplicación. EventVerse, se reserva el derecho de modificar cualquier tipo de información que pudiera aparecer en el sitio web, sin que exista obligación de preavisar o poner en conocimiento de los usuarios dichas obligaciones, entendiéndose como suficiente con la publicación en el sitio web de EventVerse
        </p>
      </p>
    </p>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Objeto
      </h1>
    </div>
    <p
      class="c3"
    >
      <p
        class="c4"
      >
        <p
          class="c5"
        >
          A través del Sitio Web, les ofrecemos a los Usuarios la posibilidad de acceder a la información sobre nuestros servicios.
        </p>
      </p>
    </p>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Privacidad y tratamiento de datos
      </h1>
    </div>
    <p
      class="c3"
    >
      <p
        class="c4"
      >
        <p
          class="c5"
        >
          Cuando para el acceso a determinados contenidos o servicio sea necesario facilitar datos de carácter personal, los Usuarios garantizarán su veracidad, exactitud, autenticidad y vigencia. La empresa dará a dichos datos el tratamiento automatizado que corresponda en función de su naturaleza o finalidad, en los términos indicados en la sección de Política de Privacidad.
        </p>
      </p>
    </p>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Propiedad industrial e intelectual
      </h1>
    </div>
    <p
      class="c3"
    >
      <p
        class="c4"
      >
        <p
          class="c5"
        >
          El Usuario reconoce y acepta que todos los contenidos que se muestran en el Espacio Web y en especial, diseños, textos, imágenes, logos, iconos, botones, software, nombres comerciales, marcas, o cualesquiera otros signos susceptibles de utilización industrial y/o comercial están sujetos a derechos de Propiedad Intelectual y todas las marcas, nombres comerciales o signos distintivos, todos los derechos de propiedad industrial e intelectual, sobre los contenidos y/o cualesquiera otros elementos insertados en el página, que son propiedad exclusiva de la empresa y/o de terceros, quienes tienen el derecho exclusivo de utilizarlos en el tráfico económico.
        </p>
        <p
          class="c5"
        >
          Por todo ello el Usuario se compromete a no reproducir, copiar, distribuir, poner a disposición o de cualquier otra forma comunicar públicamente, transformar o modificar tales contenidos manteniendo indemne a la empresa de cualquier reclamación que se derive del incumplimiento de tales obligaciones. En ningún caso el acceso al Espacio Web implica ningún tipo de renuncia, transmisión, licencia o cesión total ni parcial de dichos derechos, salvo que se establezca expresamente lo contrario.
        </p>
        <p
          class="c5"
        >
          Las presentes Condiciones Generales de Uso del Espacio Web no confieren a los Usuarios ningún otro derecho de utilización, RRHH, alteración, explotación, reproducción, distribución o comunicación pública del Espacio Web y/o de sus Contenidos distintos de los aquí expresamente previstos. Cualquier otro uso o explotación de cualesquiera derechos estará sujeto a la previa y expresa autorización específicamente otorgada a tal efecto por la empresa o el tercero titular de los derechos afectados.
        </p>
        <p
          class="c5"
        >
          Los contenidos, textos, fotografías, diseños, logotipos, imágenes, programas de ordenador, códigos fuente y, en general, cualquier creación intelectual existente en este Espacio, así como el propio Espacio en su conjunto, como obra artística multimedia, están protegidos como derechos de autor por la legislación en materia de propiedad intelectual. La empresa es titular de los elementos que integran el diseño gráfico del Espacio Web, lo menús, botones de navegación, el código HTML, los textos, imágenes, texturas, gráficos y cualquier otro contenido del Espacio Web o, en cualquier caso dispone de la correspondiente autorización para la utilización de dichos elementos. El contenido dispuesto en el Espacio Web no podrá ser reproducido ni en todo ni en parte, ni transmitido, ni registrado por ningún sistema de recuperación de información, en ninguna forma ni en ningún medio, a menos que se cuente con la autorización previa, por escrito, de la citada Entidad. Asimismo queda prohibido suprimir, eludir y/o manipular el «copyright» así como los dispositivos técnicos de protección, o cualesquiera mecanismos de información que pudieren contener los contenidos. El Usuario de este Espacio Web se compromete a respetar los derechos enunciados y a evitar cualquier actuación que pudiera perjudicarlos, reservándose en todo caso la empresa el ejercicio de cuantos medios o acciones legales le correspondan en defensa de sus legítimos derechos de propiedad intelectual e industrial.
        </p>
      </p>
    </p>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Protección de datos
      </h1>
    </div>
    <p
      class="c3"
    >
      <p
        class="c4"
      >
        <p
          class="c5"
        >
          Para utilizar algunos de los Servicios, el Usuario debe proporcionar previamente ciertos datos de carácter personal. La empresa tratará automatizadamente estos datos y aplicará las correspondientes medidas de seguridad, todo ello en cumplimiento del RGPD, LOPDGDD y LSSI. El Usuario puede acceder a la política seguida en el tratamiento de los datos personales, así como el establecimiento de las finalidades previamente establecidas, en las condiciones definidas en la Política de Privacidad.
        </p>
      </p>
    </p>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Declaraciones y garantías
      </h1>
    </div>
    <p
      class="c3"
    >
      <p
        class="c4"
      >
        <p
          class="c5"
        >
          En general, los contenidos y servicios ofrecidos en el Espacio Web tienen carácter meramente informativo. Por consiguiente, al ofrecerlos, no se otorga garantía ni declaración alguna en relación con los contenidos y servicios ofrecidos en el Espacio web, incluyendo, a título enunciativo, garantías de licitud, fiabilidad, utilidad, veracidad, exactitud, o comerciabilidad, salvo en la medida en que por ley no puedan excluirse tales declaraciones y garantías.
        </p>
      </p>
    </p>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Fuerza mayor
      </h1>
    </div>
    <p
      class="c3"
    >
      <p
        class="c4"
      >
        <p
          class="c5"
        >
          La empresa no será responsable en todo en caso de imposibilidad de prestar servicio, si ésta se debe a interrupciones prolongadas del suministro eléctrico, líneas de telecomunicaciones, conflictos sociales, huelgas, rebelión, explosiones, inundaciones, actos y omisiones del Gobierno, y en general todos los supuestos de fuerza mayor o de caso fortuito.
        </p>
      </p>
    </p>
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Resolución de controversias. Ley aplicable y jurisdicción
      </h1>
    </div>
    <p
      class="c3"
    >
      <p
        class="c4"
      >
        <p
          class="c5"
        >
          Las presentes Condiciones Generales de Uso, así como el uso del Espacio Web, se regirán por la legislación española. Para la resolución de cualquier controversia las partes se someterán a los Juzgados y Tribunales del domicilio social del Responsable del sitio web.
        </p>
        <p
          class="c5"
        >
          En el supuesto de que cualquier estipulación de las presentes Condiciones Generales de Uso resultara inexigible o nula en virtud de la legislación aplicable o como consecuencia de una resolución judicial o administrativa, dicha inexigibilidad o nulidad no hará que las presentes Condiciones Generales de Uso resulten inexigibles o nulas en su conjunto. En dichos casos, la empresa procederá a la modificación o sustitución de dicha estipulación por otra que sea válida y exigible y que, en la medida de lo posible, consiga el objetivo y pretensión reflejados en la estipulación original.
        </p>
      </p>
    </p>
  </div>
</div>
`;
